### ISSUE TITLE
[L] View Protected Area Search Result List

### ISSUE THREAD

**Issue by ashley-dhillon** [bcgov/parks-data-register#27]
As a park name registrar, I need to view a list of protected area names by ParkID (aka ORCS) so that I can easily find what I am looking for in the list.

**Updated MVP AC:**

~~- View complete list of names organized by ORC/ID number~~
- List respects search criteria aka shows results
- Search results table shows ID, Legal Name, Type, Protected Area Status and Actions
- UI matches designs

Filtering by Status
- Search for the keyword "park" and along with that try applying filters from the "Filter by protected area status"

~~**View all records search results**~~
~~If the user clicks “Search” with the default filters/searches as shown, a full list of all records will be returned.~~

~~**Load more button**~~
~~Return 20 search results per batch. If more than 20 search results, display Load more button. On click, load 20 more search results.~~

Removed the above requirement as it was identified as a dev risk and not a strong need for the user to do this. Will reassess need and design should users request it after launch.

**Notes:**
- First iteration will not include sites
- *Backlog Refinement flag - This issue was originally estimated as a L (5) but no longer includes #43 (provide URL to results) and #44 (export search results) so will need to be re-estimated.

![Image](https://github.com/bcgov/parks-data-register/assets/92062502/3a9da40b-75b8-4fc5-bd45-4e41536f3f01)
![Image](https://github.com/bcgov/parks-data-register/assets/92062502/e5972fb3-fb67-48bb-900b-cea18c12258a)
![Image](https://github.com/bcgov/parks-data-register/assets/92062502/6332f1ea-13a2-4911-8104-07e208a1d82b)

![Image](https://github.com/bcgov/parks-data-register/assets/92062502/b8054dec-fd72-40de-8be1-6632dfcc892b)

Figma design link

https://www.figma.com/file/qdjegD6YHWZPlyFT8dGXvk/Park-Name-Data-Register?type=design&node-id=388%3A5511&mode=design&t=inwvIWrUSKnKJIKl-1

**Comment by JLWade** [comment]
@janemoxd for status in this list, do we want it to reflect park name status? So historical or current will be the status options here (no repealed). And to find a repealed park it would show the park name status of historical.

@cameronpettit @danieltruong  fyi

**Comment by danieltruong** [comment]
Another question, from what I understand, the only way to get to design 1.1.1.1c is via the search page clicking on the edit button in the list. With these changes regarding the historical status, how will the flow to get to 1.1.1.1c change?

**Comment by janemoxd** [comment]
@JLWade That wasn't the original idea - but with statuses in flux, we'll need to wait until that's resolved to update what statuses should be in the dropdown. If we go with the concept of protected area ID as an object that can have a status, then the dropdown would include the protected area statuses.

Having name status in this dropdown wouldn't make any sense because from the user's POV, the current and historical names show up in the same place on the UI - so you couldn't search for a historical name without also viewing the current name and vice versa.

**Comment by janemoxd** [comment]
@danieltruong That's a good question. I don't think there should be any change (except for maybe the wording). But want the status question to be resolved before I can answer that for certain.

**Comment by JLWade** [comment]
Let's leave this in blocked until we sort that status piece out. 
Thanks for the responses!

**Comment by janemoxd** [comment]
Questions in this ticket are resolved. Attaching new user flow diagram to answer anything else outstanding.

**Comment by danieltruong** [comment]
Blocked by question in Dev channel.

"I am currently looking at ticket 27 design 1.0d. In the notes of this page, it states that "If the user clicks “Search” with the default filters/searches as shown, a full list of all records will be returned."  Returning every single record in the database in no particular order is an expensive query that I feel serves no business need. Is there an actual need for this query?"

**Comment by danieltruong** [comment]
The ability to get all records has been removed. Moving to in progress.

**Comment by janemoxd** [comment]
Updated AC and removed the UI relevant to 'get all records'. Discussion was had with dev and design in teams around this feature and was agreed to remove it. Will re-assess design after launch if the user need arises for this type of full list.

**Comment by danieltruong** [comment]
Blocked by #26

**Comment by JLWade** [comment]
moving back to in progress as #26 is close to completion and this can be unblocked as is being worked on as demoed at standup this morning

**Comment by manuji** [comment]
Tested on TEST: **Done**
* Testing has been completed and verified that as a user I can search for a keyword and retrieve protected area search results. 
* A separate ticket created for a bug: https://github.com/bcgov/parks-data-register/issues/295
* Passing to PO review.

**Comment by JLWade** [comment]
working for me and will prioritize that bug - moving to done
